core/test_integration.py

```diff
@@ -0,0 +1,72 @@
+from django.contrib.staticfiles.testing import StaticLiveServerTestCase
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from django.contrib.auth import get_user_model
+from django.urls import reverse
+
+from selenium import webdriver
+from selenium.webdriver.firefox.service import Service
+from webdriver_manager.firefox import GeckoDriverManager
+import time 
+    
+class TestUserLogin(StaticLiveServerTestCase):
+    '''
+    #Using ChromeDriver
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.browser = webdriver.Chrome()
+        cls.browser.implicitly_wait(10)
+        #cls.wait = WebDriverWait(cls.browser, 10)
+        cls.wait = WebDriverWait(cls.browser, 50)  # Increase wait time to 20 seconds
+    '''
+    
+
+    @classmethod
+    def setUpClass(cls):
+        # Setup code to run once before all tests, e.g., starting a browser
+        cls.browser = webdriver.Firefox(service=Service(GeckoDriverManager().install()))
+        cls.browser.implicitly_wait(10)
+        cls.wait = WebDriverWait(cls.browser, 10)
+        super().setUpClass()
+
+    @classmethod
+    def tearDownClass(cls):
+        time.sleep(5)  # Keep the browser open for 5 seconds
+        cls.browser.quit()
+        super().tearDownClass()
+
+    def setUp(self):
+        # Setup code to run before each test, e.g., creating test data
+        User = get_user_model()
+        self.user = User.objects.create_user(username='testuser', password='testpassword')
```

Comment:
I might move the creation of the user into the actual test, since we'll probably want to test the signup form in this same class.